addons/mail/static/src/new/core/thread_service.js

```diff
@@ -131,7 +131,10 @@ export class ThreadService {
                 last_message_id: mostRecentNonTransientMessage.id,
             });
         }
-        this.update(thread, { isUnread: false });
+        this.update(thread, {
+            isUnread: false,
+            serverLastSeenMsgBySelf: mostRecentNonTransientMessage?.id,
```

Comment:
update `serverLastSeenMsgBySelf` in the branching of actually marking as read to server